### ISSUE TITLE
Change to the rating system

### ISSUE THREAD

**Issue by alareth1976** [ampache/ampache#1754]
# Environment

 * Ampache version: Ampache 3.8.6-release
 * Web server + version: Apache 2.4.25-3+deb9u4
 * Server operating system: Linux 4.9.59-v7+ #1047 SMP Sun Oct 29 12:19:23 GMT 2017 armv7l GNU/Linux (Raspbian)
 
# Description
I'm a truly happy Ampache user but I think that the rating system should be modified to allow the use of half star or to expand from 5 stars to 10.
For me 5 stars are not enough to truly rate songs, albums and artists. There are too many shades of "love" (or hate :) ) that it's lost using the current rating system.
For example, keeping it simple, considering my ratings for this 4 bands

Pink Floyd 10
Carcass 9
Skyclad 9
Manowar  7

with the actual rating system it will be something like this 
Pink Floyd *****
Carcass     *****
Skyclad     *****
Manowar  ****
( 5 stars = 10-9 
  4 stars = 8-7
  3 stars = 6-5
  2 stars = 4-3
  1 star   = 2-1)
and it's clear that is not really representative of the real ratings.
I would love if you would at least consider this :)

thanks 
Dimitri

**Comment by jNullj** [comment]
How about a user settings to enable users set if they want a simple 1-5 stars rating view or more complex 1-100 for those who want a large score margin.
The db will save real rating as values between 1-100 and show them as 1-5 stars as needed.
That way each user can decide if they prefere the ease of the 5 stars system or the extra depth/control in a stronger 100 values rating.
It will probebly be easy to use a slider to vote with so many values.

If im not wrong the 5-stars rating system might be related to file's metadata, i think most players (windows file explorer as well) show and set the rating in mp3 metadata as 1-5 stars.
But the rating should actually be saved as values from 1-255 at mp3 (i learn something new everyday) which each players defines the range for each rating diffrently.
[read about tags here - rating is at 4.17](http://id3.org/id3v2.4.0-frames)

Do you think there is some sense in using 1-100, 1-255 or having the option to switch rating "views"?

**Comment by alareth1976** [comment]
> 
> 
> How about a user settings to enable users set if they want a simple 1-5 stars rating view or more complex 1-100 for those who want a large score margin.
> The db will save real rating as values between 1-100 and show them as 1-5 stars as needed.
> That way each user can decide if they prefere the ease of the 5 stars system or the extra depth/control in a stronger 100 values rating.
> It will probebly be easy to use a slider to vote with so many values.
> 
> If im not wrong the 5-stars rating system might be related to file's metadata, i think most players (windows file explorer as well) show and set the rating in mp3 metadata as 1-5 stars.
> But the rating should actually be saved as values from 1-255 at mp3 (i learn something new everyday) which each players defines the range for each rating diffrently.
> [read about tags here - rating is at 4.17](http://id3.org/id3v2.4.0-frames)
> 
> Do you think there is some sense in using 1-100, 1-255 or having the option to switch rating "views"?

I think it would be a great idea :)

**Comment by xekon** [comment]
I absolutely love this idea.

**Comment by AshotN** [comment]
Since Ampache keeps track of average rating between users, there will have to also be some logic to handle that correctly. Also how would the UI look in a 1-100 rating scenario?

This would also need API support for clients to be able to correctly handle a new rating system.

Overall it be a decent bit of work. I wouldn't count on it happening soon sadly.

**Comment by lachlan-00** [comment]
It's not something that i think would be of benefit to focus on at least for the current work i need to get done.

**Comment by xekon** [comment]
I am not sure what affect it might have changing the scale of the rating system, but by allowing half stars I would assume the currently stored values would still be valid and that it would simple allow more ganular data to be stored?

For example just changing the stored values from integer to float to allow 3.5 isntead of a value of an integer value of 3 or 4?

Clementine uses half star rating exactly like this:

![image](https://user-images.githubusercontent.com/106644/75206592-a44de500-572b-11ea-8a1d-43c38b01a4a2.png)

**Comment by lachlan-00** [comment]
half stars already exist in the averages.

**Comment by lachlan-00** [comment]
For regular ratings, i'm not sure if it's rounded (i think it is) but i'll check

**Comment by xekon** [comment]
yeah when I said "I absolutely love this idea." I should have clarified that I was talking about the original post idea of half stars, I don't actually have a need for a 1-100 rating system.

**Comment by Jieiku** [comment]
I make lots of smart playlists that use the ratings system.

Currently because there are only 5 star ratings and we cannot actually vote with half stars, this is the way I generally vote:

1star - awful, I never want to hear this again EVER.
2star - I do not really care for this song, I would prefer to not listen to it.
3star - I consider this song OK, I do not mind listening to it if other people also want to listen to it.
4star - I really like this song, its great!
5star - I love this song, its one of my favorites.

Keeping this in mind when I set song ratings, it has worked out pretty good.
